### ISSUE TITLE
Amplify Push fails with no error message

### ISSUE THREAD

**Issue by kimfucious** [aws-amplify/amplify-cli#13008]
### How did you install the Amplify CLI?

NPM

### If applicable, what version of Node.js are you using?

16

### Amplify CLI Version

10.7.3

### What operating system are you using?

macOS

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes made

### Describe the bug

I returned to an app that has been deployed w/o changes quite some time (over a year).

I modified a single lambda function and did an `amplify push`.

The process failed and an error was presented letting me know that the version of node configured for the lambdas used by my app is deprecated.

I manually upgraded all of the lambas to version 18.x and tried to push again.

There is the brief flash of a red error and then the below is presented:

```console
Deploying resources into prod environment. This will take a few minutes. ⠴
Deploying root stack thaiFormality [ ---------------------------------------- ] 0/13
Deploying auth thaiformalitye588a710 [ ---------------------------------------- ] 0/11
Deploying function thaiFormality [ ---------------------------------------- ] 0/4
Deploying resources into prod environment. This will take a few minutes. ⠇
Deploying root stack thaiFormality [ ---------------------------------------- ] 0/13
Deploying auth thaiformalitye588a710 [ ---------------------------------------- ] 0/11
Deploying function thaiFormality [ ---------------------------------------- ] 0/4
Deploying function thaiFormalitySendEmail [ ---------------------------------------- ] 0/4
Deploying funDeploying resources into prod environment. This will take a few minutes. ⠹
Deploying root stack thaiFormality [ ---------------------------------------- ] 0/13
Deploying auth thaiformalitye588a710 [ ---------------------------------------- ] 0/11
Deploying function thaiFormality [ ---------------------------------------- ] 0/4
Deploying function thaiFormalitySendEmail [ ---------------------------------------- ] 0/4
Deploying function thaiFormalityCognito [ ---------------------------------------- ] 0/4
Deploying function thaiPostalCodeLookup [ ---------------------------------------- ] 0/3
Deploying function thaiFormalityAddressLookupDemo [ ---------------------------------------- ] 0/4
Deploying function thaiFormalityFormDemo [ ---------------------------------------- ] 0/4
Deploying function thaiFormalityPaypal [ ---------------------------------------- ] 0/3
Deploying function thaiFormalityPaypalToken [ ---------------------------------------- ] 0/4
Deploying api thaiFormalityAPI [ ---------------------------------------- ] 0/12
Deploying api thaiformalityGql [ ---------------------------------------- ] 0/9
Deploying storage thaiFormalityS3 [ ---------------------------------------- ] 0/9
✔ Done

Project Identifier: bd73bb1bbce3de31d55470e668235b85
```
I cannot determine anything from the logs except that some of the nested stacks in Cloud Formation (i.e. the lambdas) are in a failed state.

Looking in CloudFormation, the lambdas are showing like this:
![image](https://github.com/aws-amplify/amplify-cli/assets/26792769/b3764f96-a69d-4fd8-a54c-b122d498b202)

I've tried to run `continue update rollback` from the root, but that didn't resolve the issue.

Any advice on how to resolve this would be most appreciated.

### Expected behavior

Amplify push should work after having updated the lambda functions to v. 18x.

Cloud Formation rollback should complete w/o errors or there should be a way push them forward to a working state.

### Reproduction steps

1.  Have a nice working app.
2. Wait a long time before touching it.
3. Update a lambda.
4. Run `amplify push`
5. Watch it break.

### Project Identifier

bd73bb1bbce3de31d55470e668235b85

### Log output

<details>

```
# Put your logs below this line

```
</details>

### Additional information

I've attempted to follow the instructions laid out [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-continueupdaterollback.html) to perform a `continue update rollback`. 

I skipped the lambdas with `UPDATE_ROLLBACK_FAILED` status under `Advanced troubleshooting`.

Under events, I'm presented with this error:

```console
Nested stack resources can only be skipped when their embedded stack statuses are one of [DELETE_COMPLETE, DELETE_IN_PROGRESS, DELETE_FAILED]
```

At this point, I am at a loss regarding how to proceed.

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by kimfucious** [comment]
I kinda figured this out, but I'm leaving this open in case there's a better way to do what I did.

In short, I followed the instructions [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-continueupdaterollback.html) to perform a `continue update rollback`.

I expanded the `Advanced troubleshooting` section and instead of selecting the stacks containing the lambdas, I expanded each of them and selected the lambdas only (to skip them).  This got me to the `UPDATE_COMPLETE` stage.

Then what I did--and I'm not sure if this is the best way, but it seemed a lot easier than what was documented--I locally edited the `xxx-cloudformation-template.json` file, updating the Runtime to `nodejs18.x`.

After that, I was able to perform an `amplify push` successfully.
